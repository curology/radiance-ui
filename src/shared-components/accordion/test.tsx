import React from 'react';
import { render, userEvent } from 'src/tests/testingLibraryHelpers';

import { Accordion } from './index';

const testAccordionProps = {
  title: <div>title</div>,
  isOpen: false,
  onClick: (): void => undefined,
  children: <div>expansion</div>,
};

describe('<Accordion />', () => {
  /**
   * TODO: Fix Emotion 11 CI snapshot serializer order issue
   */
  // eslint-disable-next-line jest/no-disabled-tests
  describe('UI snapshots', () => {
    it('renders regular accordion', () => {
      const { container } = render(<Accordion {...testAccordionProps} />);

      expect(container.firstElementChild).toMatchSnapshot();
    });

    it('renders no border accordion', () => {
      const { container } = render(
        <Accordion {...testAccordionProps} noBorder />,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });

    it('renders disabled accordion', () => {
      const { container } = render(
        <Accordion {...testAccordionProps} disabled />,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });
  });

  it('invokes onClick when title is clicked', () => {
    const spy = jest.fn();

    const { getByRole } = render(
      <Accordion {...testAccordionProps} onClick={spy} />,
    );

    userEvent.click(getByRole('button'));

    expect(spy).toHaveBeenCalled();
  });
});
