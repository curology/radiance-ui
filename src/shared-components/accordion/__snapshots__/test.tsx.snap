// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> UI snapshots renders disabled accordion 1`] = `
.emotion-0 {
  border: 1px solid #EDEDF0;
  width: 100%;
  opacity: 0.4;
  background-color: #EDEDF0;
  border-color: #C3C0CD;
}

.emotion-0:not(:last-child) {
  border-bottom: none;
}

.emotion-1:last-of-type .css-sk7hsm-TitleWrapper:focus {
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
}

.emotion-1:last-of-type .emotion-2:focus {
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  cursor: not-allowed;
}

.emotion-2:focus {
  outline: none;
  box-shadow: inset 0px 0px 0px 2px #332E54;
}

.emotion-4 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  width: 100%;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1rem;
}

.emotion-8 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  color: #332E54;
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-9 {
  max-height: 0px;
  overflow: hidden;
  -webkit-transition: max-height 350ms ease-in-out;
  transition: max-height 350ms ease-in-out;
}

<div
  class="emotion-0 emotion-1"
  disabled=""
>
  <div
    aria-disabled="true"
    aria-expanded="false"
    class="emotion-2 emotion-3"
    disabled=""
    role="button"
    tabindex="-1"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <div>
        title
      </div>
    </div>
    <div
      class="emotion-6 emotion-7"
    >
      <svg
        class="emotion-8"
        fill="#332E54"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Chevron Icon
        </title>
        <path
          d="M6.87999 12.88L6 12L9.55 8.44L6 4.88L6.87999 4L11.32 8.44L6.87999 12.88Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <div
    aria-disabled="true"
    aria-hidden="true"
    class="emotion-9 emotion-10"
  >
    <div>
      <div>
        expansion
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> UI snapshots renders no border accordion 1`] = `
.emotion-0 {
  width: 100%;
}

.emotion-0:not(:last-child) {
  border-bottom: none;
}

.emotion-1:last-of-type .emotion-2:focus {
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  cursor: pointer;
}

.emotion-2:focus {
  outline: none;
  box-shadow: inset 0px 0px 0px 2px #332E54;
}

.emotion-4 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  width: 100%;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1rem;
}

.emotion-8 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  color: #332E54;
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-9 {
  max-height: 0px;
  overflow: hidden;
  -webkit-transition: max-height 350ms ease-in-out;
  transition: max-height 350ms ease-in-out;
}

<div
  class="emotion-0 emotion-1"
>
  <div
    aria-disabled="false"
    aria-expanded="false"
    class="emotion-2 emotion-3"
    role="button"
    tabindex="0"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <div>
        title
      </div>
    </div>
    <div
      class="emotion-6 emotion-7"
    >
      <svg
        class="emotion-8"
        fill="#332E54"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Chevron Icon
        </title>
        <path
          d="M6.87999 12.88L6 12L9.55 8.44L6 4.88L6.87999 4L11.32 8.44L6.87999 12.88Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <div
    aria-disabled="false"
    aria-hidden="true"
    class="emotion-9 emotion-10"
  >
    <div>
      <div>
        expansion
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> UI snapshots renders regular accordion 1`] = `
.emotion-0 {
  border: 1px solid #EDEDF0;
  width: 100%;
}

.emotion-0:not(:last-child) {
  border-bottom: none;
}

.emotion-1:last-of-type .emotion-2:focus {
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  cursor: pointer;
}

.emotion-2:focus {
  outline: none;
  box-shadow: inset 0px 0px 0px 2px #332E54;
}

.emotion-4 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  width: 100%;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1rem;
}

.emotion-8 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  color: #332E54;
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-9 {
  max-height: 0px;
  overflow: hidden;
  -webkit-transition: max-height 350ms ease-in-out;
  transition: max-height 350ms ease-in-out;
}

<div
  class="emotion-0 emotion-1"
>
  <div
    aria-disabled="false"
    aria-expanded="false"
    class="emotion-2 emotion-3"
    role="button"
    tabindex="0"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <div>
        title
      </div>
    </div>
    <div
      class="emotion-6 emotion-7"
    >
      <svg
        class="emotion-8"
        fill="#332E54"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Chevron Icon
        </title>
        <path
          d="M6.87999 12.88L6 12L9.55 8.44L6 4.88L6.87999 4L11.32 8.44L6.87999 12.88Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <div
    aria-disabled="false"
    aria-hidden="true"
    class="emotion-9 emotion-10"
  >
    <div>
      <div>
        expansion
      </div>
    </div>
  </div>
</div>
`;
